src/pom.xml

```
@@ -2374,7 +2384,40 @@
           </execution>
         </executions>
       </plugin>
-
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-enforcer-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>enforce-maven-version</id>
+            <!-- Enforcing a minimum Maven version is crucial to ensure that:
+                 - Plugin version resolution works as expected: Different Maven versions
+                   may resolve plugin versions differently due to changes in the default
+                   lifecycle mappings, dependency resolution behavior, and plugin inheritance.
+                 - Features and bug fixes required by newer plugins are available: For example,
+                   some plugins (e.g., surefire-plugin 3.x) require Maven 3.6+ or 3.8+.
+                 - Consistent builds across environments: Without enforcing a Maven version,
+                   developers or CI/CD pipelines using different Maven versions may face
+                   unexpected build failures or inconsistencies.
+            -->
+            <goals>
+              <goal>enforce</goal>
+            </goals>
+            <configuration>
+              <rules>
+                <requireMavenVersion>
+                  <version>3.9+</version>
```

While this may work, it's not in line with [documentation](https://cwiki.apache.org/confluence/display/MAVENOLD/Dependency+Mediation+and+Conflict+Resolution#DependencyMediationandConflictResolution-DependencyVersionRanges) and [versionRanges](https://maven.apache.org/enforcer/enforcer-rules/versionRanges.html) which suggests this should be 

```suggestion
                  <version>[3.9,)</version>
```

that would however suggest the upcoming version 4 should work which is unlikely, so better would be 
```suggestion
                  <version>[3.9,4.0)</version>
```
`3.9 <= x < 4.0` which excludes 4

---

good catch. Works though, but who knows what the semantics really are. Seems to be common in gradle

---

[done](https://github.com/geoserver/geoserver/compare/3901b16eaede993eedce9ed989fd24c0768ce187..9524bfb57fdcabb4ed45e3ce0eeb739b2af5c380)